Add introduction and conclusion sections

MichalisPanayides opened the pull request:
- [x] Introduction
   - [x] General intro
   - [x] Background info using news reports
   - [x] Literature review: Game theory & Queueing theory
   - [x] Literature review: Multi-network queues
   - [x] Establish the research problem and specify objectives
   - [x] Map of the paper
- [x] Conclusion
   - [x] Paper overview
   - [x] Objectives
   - [x] Limitations
   - [x] Future work

MichalisPanayides committed:
Update author names

MichalisPanayides committed:
Update bibliography

Add papers that combine game theory and queueing theory

MichalisPanayides committed:
Add initial draft of introduction

- First line (hook)
- Literature review methodology driven (game theory - queueing theory)
- Establish research problem
- Research objectives
- Map out the paper

MichalisPanayides committed:
Cite papers on networks with tandem queues

MichalisPanayides committed:
Remove unused citation

MichalisPanayides committed:
Add some additional references

MichalisPanayides committed:
Minor changes

- Remove unnecessary comments
- Apply suggestions based on alex and proselint
- Reference a paper on ED pressures
- Reference paper on 2-dimensional Markov model with game theory
- Clearly state the objective and the "so what?"
- Other minor changes

MichalisPanayides committed:
Update known words

MichalisPanayides committed:
Include more references of news reports

MichalisPanayides committed:
Some minor changes

- Discuss cited news report a bit more
- Rephrase one of the objectives
- Other minor changes

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -1 +1,97 @@
\ No newline at end of file
+\section{Introduction}
+
+A huge issue for ambulances in the United Kingdom is that they stay 
```
```suggestion
An important issue for ambulances in the United Kingdom is that they stay 
```

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -1 +1,97 @@
+\section{Introduction}
+
+A huge issue for ambulances in the United Kingdom is that they stay 
+parked outside hospitals for a huge amount of time. 
```
I feel a reviewer might ask for some sort of evidence of this. What would be great would be a reference in a journal where they analysed waiting of parked ambulances. If we cannot find that, let us put in one of the many news articles.

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -1 +1,97 @@
+queueing model motivated by this scenario.
+
+Emergency departments (EDs) are constantly under a lot of pressure to meet 
+targets and satisfy regulations \cite{EmergencyDepartmentWinterPressures}.
```
```suggestion
targets and satisfy regulations~\cite{EmergencyDepartmentWinterPressures}.
```

This creates a space that won't be broken at a line. So you will never get a new line that just started with a reference.

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -1 +1,97 @@
+issue that affects both the patients and the emergency medical services (EMS).
+Due to this blockage patients may spend a considerable amount of time in the 
+back of an ambulance waiting to be dispatched to the ED.
+Examples of such reports are \cite{mirror}, \cite{thenews} and \cite{bmj} where
```
```suggestion
Examples of such reports are \cite{mirror, thenews, bmj} where
```

MichalisPanayides replied:
I did not know you can do this. Nice 👍

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -1 +1,97 @@
+By blocking the ambulances at the hospital's parking space, valuable time is 
+wasted where ambulances could be attending new patients \cite{eastanglia}.
+
+In the past a number of papers have been published that touch upon the use of 
```
```suggestion
A number of papers have been published that touch upon the use of 
```

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -1 +1,97 @@
+
+In the past a number of papers have been published that touch upon the use of 
+queueing models together with game theoretic concepts.
+In \cite{FirmCompetition} the authors study a simultaneous price competition 
```
This reference seems out of place as is. In the sentence above you mention how a number of papers are at the intersection of queueing models and game theory. Is this one of them?

MichalisPanayides replied:
Yes, this is one of them. Should I expand more on the queueing part?

drvinceknight replied:
Yes I think so, as is what you have written is a description of any paper where there are two players in it.

MichalisPanayides replied:
Yep, I see what you are saying.

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -1 +1,97 @@
+In \cite{knight2017measuring} a normal form game is built that is informed by a 
+two-dimensional Markov chain in order to model interactions between critical
+care units.
+An additional paper that focuses on competition is \cite{fan2009short} where
```
The link to queues is not clear here.

MichalisPanayides replied:
Similar to the previous comment ☝️. Should I expand on the queueing part of these?

By the way I got most of these from the book `Rational Queueing`

drvinceknight replied:
Yup, just need to expand/clarify why the is relevant to queuing.

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -1 +1,97 @@
+different implementation costs.
+In \cite{sadat2015can} a healthcare application was studied where patients 
+could choose between two hospitals, where a utility function is derived that is
+based on patients' perceived quality of life.
```
You should add the [PoA](https://www.sciencedirect.com/science/article/pii/S0377221713003019).

There are also a fair few healthcare ones that I think you've omitted (the ambulance diversion one specifically).

MichalisPanayides replied:
Will do

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -1 +1,97 @@
+with two tandem queues.
+In \cite{d2015pure} the authors explore threshold joining strategies in a 
+Markov model that has two tandem queues.
+Another great example is the one described in \cite{burnetas2013customer}
```
Tell me about this paper. (No need to respond, just one that I'd like us to discuss.)

MichalisPanayides replied:
`\cite{burnetas2013customer}` This one right?

drvinceknight replied:
Yup.

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -1 +1,97 @@
+Another great example is the one described in \cite{burnetas2013customer}
+where they investigated a network of multiple tandem queues where customers 
+decide which queue to attend before joining.
+Similarly, in \cite{bacsar2002stackelberg} the authors examine a network of 
```
Same as above, are you sure it's tandem?

MichalisPanayides replied:
That's how the authors define it at the `problem formulation` section of that paper and that's also how `Hassin, R.` introduces it in `Rational Queueing`.

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -1 +1,97 @@
+maximise utility.
+In \cite{veltman2005equilibrium} a profit maximisation problem is studied that
+has an M/M/1 queue and a parking service providing complementary service while
+the customer is in service. 
```
What do they do?

MichalisPanayides replied:
Should I include more details on the methodology that has been used here?

drvinceknight replied:
Just a bit but also perhaps the conclusions.

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -1 +1,97 @@
+the customer is in service. 
+The problem was later extended by \cite{sun2009equilibrium} where they 
+considered arrivals of batches that can share the parking service.
+Finally, \cite{afeche2007decentralized} examines a tandem network of two M/M/1 
```
Same comment as above

MichalisPanayides replied:
Same reply as above 😄

drvinceknight replied:
Same reply.

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -1 +1,97 @@
+based on patients' perceived quality of life.
+
+Another specific part of this research is the construction of a queueing system
+with two tandem queues.
```
Let us say here what we mean by tandem queues. Arguably your problem is not a tandem queue.

MichalisPanayides replied:
Isn't it kind of a tandem queue though? Can I not label the model as a `network of tandem queues`?

drvinceknight replied:
Yup it's absolutely a kind of tandem queue but it's really a tandem buffer with a **single** queue.

MichalisPanayides replied:
Hmmmm. Should I just say that then?

```suggestion
with two tandem buffers.
```

drvinceknight replied:
I don't know. I'm not actually sure what the best description of this is as to my knowledge it's the first time this has been done. Perhaps just what it is: `a tandem buffer with a single service centre`?

MichalisPanayides replied:
Sounds good. Will go with that for now then 👍

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -1 +1,97 @@
+receive three different types of customers.
+
+The EDs in the United Kingdom have to follow some set of regulations imposed to 
+them by the NHS.
```
```suggestion
them by the National Health Service (NHS).
```

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -1 +1,97 @@
+be admitted, transferred or discharged within four hours.
+This is where gaming behaviour might be observed between the EDs and the EMS.
+An assumption of this work is that some managerial decision making is involved
+in choosing when to start blocking ambulances.
```
`This is similar to~\cite{<that ambulance diversion paper>}...`

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -1 +1,97 @@
+to them. 
+The resultant model will then be used to explore the emergent dynamics between 
+the three players.
+The model is then applied to the healthcare scenario between two EDs and the 
```
Highlight how this differs from what's in the literature.

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -1 +1,97 @@
+follows with the methodology that was used to build the game.
+Finally, there is a section that describes how this game theoretic framework
+can be applied at the interface between the EMS and EDs together with some 
+analysis on the behaviour that is observed.
```
```suggestion
This paper first gives an overview of the game theoretic model, 
then goes on to describe a novel queueing model.
Finally, there is a section that describes how this game theoretic framework
can be applied at the interface between the EMS and EDs together with some 
analysis on the behaviour that is observed.
```

MichalisPanayides replied:
Should I not mention the methodology section at all here?

drvinceknight replied:
No sorry, you should I removed that by mistake.

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -1 +1,97 @@
+the three players.
+The model is then applied to the healthcare scenario between two EDs and the 
+EMS while looking into the inefficiencies that emerge and ways to apply some 
+incentive mechanisms to improve them.
```
Be a bit louder here about the queueing model.

MichalisPanayides committed:
Apply minor suggestions from code review

Co-authored-by: Vince Knight <vincent.knight@gmail.com>

MichalisPanayides committed:
Add additional changes based on PR review

- Add more citations
- Explain further some of the references
- Other minor changes

MichalisPanayides committed:
Apply minor changes

- Changes based on proselint
- Changes based on alex
- State gap in literature

MichalisPanayides committed:
Update known words

drvinceknight commented on bibliography.bib:
```diff
@@ -309,6 +309,48 @@ @article{knight2017measuring
+
 ###################
 ## NEWS ARTICLES ##
 ###################
```
I believe `s/#/%` is more robust here.

MichalisPanayides replied:
What do you mean? Should I remove that comment? Or should I just say:

`# NEWS ARTICLES`

drvinceknight replied:
No I believe technically you're using `#` for comments which could be brittle for the TeX compiler which would look for `%` as a comment.

MichalisPanayides replied:
I think the .bib file doesn't support comments (might be wrong though). You can type anything outside of environment and wouldn't make a difference:

```
@article{key
   ...
}
```

But I guess I can use `%` for consistency

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -15,54 +17,82 @@ \section{Introduction}
+a product instantly and seller 2 is modelled as a make-to-order M/M/1 queue.
+The game that is played requires the two sellers to make a choice on the price 
+of the product and then seller 2 to set a capacity that guarantees a maximum 
+expected delay.
+In~\cite{sadat2015can} a healthcare application was studied where patients 
```
Something like:

```
In the above models, the players are attempting to increase 
their share of individuals choosing to queue.
In public healthcare type settings~\cite{knight2013selfish}, 
this is not necessarily the case. Players might not necessarily want and/or need 
to maximise so called `market share' and overall service needs to be considered as 
individuals aim to minimise experience congestion.
```

MichalisPanayides replied:
I will put this before the healthcare applications

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -15,54 +17,82 @@ \section{Introduction}
+concept known as the price of anarchy (\(PoA\))~\cite{koutsoupias1999worst}.
+They show that the price of anarchy increases with worth of service and that is
+low for systems with insufficient capacities.
+In \cite{deo2011centralized} the authors study the network effect of ambulance 
```
`In~\cite{...`

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -15,54 +17,82 @@ \section{Introduction}
+efficiency of public services is investigated.
+The authors place the individuals' choices between different public services 
+within the formulation of routing games and measure inefficiencies using a 
+concept known as the price of anarchy (\(PoA\))~\cite{koutsoupias1999worst}.
```
`\(\text{PoA}\)` otherwise it looks like `P \times o \times A`.

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -15,54 +17,82 @@ \section{Introduction}
+In~\cite{sadat2015can} a healthcare application was studied where patients 
 could choose between two hospitals, where a utility function is derived that is
 based on patients' perceived quality of life.
+In~\cite{knight2013selfish} the impact that individual behaviour has on the 
```
`How individual behaviour based on congestion impacts the efficiency of public services is investigated in~\cite{knight2013selfish}.`

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -15,54 +17,82 @@ \section{Introduction}
+customers requiring the first service and customers requiring the second service.
+The authors showed that optimal prices also maximise social welfare and that
+removing two types of customers that don't need both services leads to higher 
+profit and lower demand rate.
```
```
The novelty of the queueing model here is a contribution to 
not only the game theoretic literature but also to the queueing theoretic literature.
To the authors knowledge, no such model of a tandem queueing model controlled 
by a strategic decision has been considered. % TODO Be very sure if this is correct 
```

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -82,16 +113,18 @@ \section{Introduction}
 The resultant model will then be used to explore the emergent dynamics between 
 the three players.
+The new thing here is the use of a learning algorithm to get to the Nash 
+equilibrium.
```
```
While giving some consideration to equilibrium behaviour similarly to the 
work of~\cite{<a number of the paper you explored above>}`, emergent behaviour 
is more precisely addressed by considering a number of learning algorithms.
```

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -82,16 +113,18 @@ \section{Introduction}
 EMS while looking into the inefficiencies that emerge and ways to apply some 
 incentive mechanisms to improve them.
+The EDs are modelled as two queueing systems each with a tandem buffer and a 
+service centre where their performance measures of the queueing models are then
+used as the utilities of the game.
```
`service centre. The performance measures ...`

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -82,16 +113,18 @@ \section{Introduction}
-then goes on to describe how the queueing models were constructed and then 
-follows with the methodology that was used to build the game.
+This paper first gives an overview of the game theoretic model, then goes on 
+to describe a novel queueing model, and then follows with the methodology that 
+was used to build the game.
```
```
and then follows with a number of theoretic results 
pertaining to the performance measures of the queuing model. 
These are used to build the utilities of the game.
```

drvinceknight reviewed (changes requested):
I'm commenting on specific commits so cannot do this as suggestions.

MichalisPanayides committed:
Further changes based on PR review

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -113,20 +122,30 @@ \section{Introduction}
+The novelty of the queueing model here is a contribution to not only the game 
+theoretic literature but also to the queueing theoretic literature.
+To the authors knowledge, no such model of a tandem queueing model controlled 
+by a strategic decision has been considered. 
```
```suggestion
To the authors knowledge, no such model of a tandem queueing model with a pair of parameters for the buffer has been considered.
```

I think?

drvinceknight commented on sections/introduction/main.tex:
```diff
@@ -48,23 +48,32 @@ \section{Introduction}
+necessarily the case. 
+Players might not necessarily aim to maximise the so called `market share'.
+Rather, the overall service needs to be considered as players aim to minimise
+the overall congestion.
```
```suggestion
their experienced congestion.
```

MichalisPanayides committed:
Moar changes based on PR review

MichalisPanayides committed:
Write conclusion section

- Write the conclusion section in conclusion/main.tex
- Also add labels to healthacre application section

MichalisPanayides committed:
Update known.py

drvinceknight commented on sections/conclusion/main.tex:
```diff
@@ -1 +1,57 @@
+stay blocked outside of the ED at the hospital's parking space in an attempt
+to satisfy these regulations.
+
+This study explores a 3-player game theoretic model between the decision makers 
```
```suggestion
This study explores a generic 3-player game theoretic model between the decision makers 
```

drvinceknight commented on sections/conclusion/main.tex:
```diff
@@ -1 +1,57 @@
+\ref{sec:queueing_model}).
+Furthermore, the formulas for the performance measures of the queueing model 
+are also derived (sections \ref{sec:waiting_time}, \ref{sec:blocking_time}, 
+\ref{sec:proportion_within_target}). 
```
```suggestion
\ref{sec:proportion_within_target}). This novel queuing model is the first contribution of the paper.
```

drvinceknight commented on sections/conclusion/main.tex:
```diff
@@ -1 +1,57 @@
+The former relates to the results of asymmetric replicator dynamics that showed 
+that inefficient scenarios can arise by playing the game while the latter 
+implies that the learned inefficiencies can be escaped by carefully applying 
+certain incentives to the players.
```
```suggestion
certain incentives to the players. 
This is applied game theoretic model is the second main contribution of this paper.
```

drvinceknight commented on sections/conclusion/main.tex:
```diff
@@ -1 +1,57 @@
+implies that the learned inefficiencies can be escaped by carefully applying 
+certain incentives to the players.
+
+This work also has certain limitations that need to addressed.
```
```suggestion
```

drvinceknight commented on sections/conclusion/main.tex:
```diff
@@ -1 +1,57 @@
+certain incentives to the players.
+
+This work also has certain limitations that need to addressed.
+The model presented here only assumes the presence of two players that can
```
```suggestion
The model presented here assumes the presence of two players that can
```

drvinceknight reviewed (changes requested):
This looks good @11michalis11 :+1:

MichalisPanayides committed:
Apply changes based on code review
